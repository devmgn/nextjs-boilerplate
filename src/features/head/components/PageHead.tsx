import Head from 'next/head';
import { APP_NAME, DEFAULT_DESCRIPTION, TITLE_SEPARATOR } from '../config';
import type { SetOptional } from 'type-fest';

type HeadProps = {
  title?: string;
  description?: string;
} & SetOptional<React.ComponentProps<typeof Head>, 'children'>;

const PageHead: React.FC<HeadProps> = ({ title, description, ...props }) => {
  return (
    <Head {...props}>
      <title>{title ? `${title}${TITLE_SEPARATOR}${APP_NAME}` : APP_NAME}</title>
      <meta name="description" content={description ?? DEFAULT_DESCRIPTION} />
    </Head>
  );
};

export default PageHead;
