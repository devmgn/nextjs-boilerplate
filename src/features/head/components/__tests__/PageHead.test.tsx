import { render } from '@testing-library/react';
import PageHead from '../PageHead';

describe('PageHead', () => {
  describe('スナップショットテスト', () => {
    test('propsが全てある場合', () => {
      const props = {
        title: 'title',
        description: 'description',
      };

      render(
        <PageHead {...props}>
          <meta name="keywords" content="keywords" />
        </PageHead>,
        { container: document.head }
      );
      expect(document.head).toMatchSnapshot();
    });

    test('propsが全てない場合', () => {
      render(<PageHead />, { container: document.head });
      expect(document.head).toMatchSnapshot();
    });

    test('propsにtitleがない場合', () => {
      const props = {
        description: 'description',
      };
      render(
        <PageHead {...props}>
          <meta name="keywords" content="keywords" />
        </PageHead>,
        { container: document.head }
      );
      expect(document.head).toMatchSnapshot();
    });

    test('propsにdescriptionがない場合', () => {
      const props = {
        title: 'title',
      };
      render(
        <PageHead {...props}>
          <meta name="keywords" content="keywords" />
        </PageHead>,
        { container: document.head }
      );
      expect(document.head).toMatchSnapshot();
    });

    test('propsにchildrenがない場合', () => {
      const props = {
        title: 'title',
        description: 'description',
      };

      render(<PageHead {...props} />, { container: document.head });
      expect(document.head).toMatchSnapshot();
    });
  });
});
