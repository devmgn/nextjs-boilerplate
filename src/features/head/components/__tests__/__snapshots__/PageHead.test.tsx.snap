// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHead スナップショットテスト propsが全てある場合 1`] = `
<head>
  <title>
    title | app name [TEST]
  </title>
  <meta
    content="description"
    name="description"
  />
</head>
`;

exports[`PageHead スナップショットテスト propsが全てない場合 1`] = `
<head>
  <title>
    app name [TEST]
  </title>
  <meta
    content="default description [TEST]"
    name="description"
  />
</head>
`;

exports[`PageHead スナップショットテスト propsにchildrenがない場合 1`] = `
<head>
  <title>
    title | app name [TEST]
  </title>
  <meta
    content="description"
    name="description"
  />
</head>
`;

exports[`PageHead スナップショットテスト propsにdescriptionがない場合 1`] = `
<head>
  <title>
    title | app name [TEST]
  </title>
  <meta
    content="default description [TEST]"
    name="description"
  />
</head>
`;

exports[`PageHead スナップショットテスト propsにtitleがない場合 1`] = `
<head>
  <title>
    app name [TEST]
  </title>
  <meta
    content="description"
    name="description"
  />
</head>
`;
